src/VisualStudio/Core/Def/Implementation/Options/LocalUserRegistryOptionPersister.cs

```diff
@@ -35,8 +35,11 @@ private LocalUserRegistryOptionPersister(RegistryKey registryKey)
         public static async Task<LocalUserRegistryOptionPersister> CreateAsync(IAsyncServiceProvider provider)
         {
             // SLocalRegistry service is free-threaded -- see https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1408594.
-            var localRegistry = await provider.GetServiceAsync<SLocalRegistry, ILocalRegistry4>(throwOnFailure: true).ConfigureAwait(false);
-            Contract.ThrowIfFalse(ErrorHandler.Succeeded(localRegistry!.GetLocalRegistryRootEx((uint)__VsLocalRegistryType.RegType_UserSettings, out var rootHandle, out var rootPath)));
+            // Note: not using IAsyncServiceProvider.GetServiceAsync<TService, TInterface> since the extension method might switch to UI thread.
+            // See https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1408619/.
+            var localRegistry = (ILocalRegistry4?)await provider.GetServiceAsync(typeof(SLocalRegistry)).ConfigureAwait(false);
+            Contract.ThrowIfNull(localRegistry);
```

Comment:
```suggestion
            Assumes.Present(localRegistry);
```

Reply:
wha'ts the difference between these two?

Reply:
`Assumes.Present` throws a better error message (includes the expected type name).

Reply:
Should we update the message Contract.ThrowIfNull reports then?

Reply:
that would be nice IMO.  I'm not a fan that there are seemingly two ways to do the exact same thing.